deploy/app/deploy-nginx.yml

```diff
@@ -2,20 +2,44 @@
 - name: Setup nginx for the application
   hosts: app
   become: true
-  become_user: root
   gather_facts: False
+  vars_prompt:
+    - name: run_roles
+      prompt: "Install nginx from scratch?"
+      default: "n"
   vars_files:
     - "vars/main.yml"
+  roles:
+    - { role: nginx, when: run_roles == "y" }
   tasks:
+    - name: Assures /etc/nginx/ssl exists
+      file: path=/etc/nginx/ssl state=directory
+      when: server_setup_ssl
+
+    - name: Copy ssl certificate to /etc/nginx/ssl
+      copy: src=ssl-keys/{{item.src}} dest={{item.dest}} owner=root mode=744
+      with_items:
+        - {src: 'app.crt', dest: '{{ssl_cert_path}}'}
```

Comment:
We copy `app.crt`, but in the nginx config below we use:
- another path (it includes domain of specific app)
- another name

Did you test it? I like an idea to autogenerate lets encrypt certificate (we does it on one of our new products, ask @KuhArt for more details). Also, aside from this: let's encrypt now supports wildcard certificates, so it is not necessary to generate one certificate per domain, we can generate wildcard.
For let's encrypt we might need to setup a separate repo, which will have: 
- guide how to create a certificate 
- some kind of ansible automation to regenerate certificate on server as it expires 

I remember @gavruk wrote an article about this on our internal blog.